riscv/triggers.cc

```diff
@@ -177,6 +177,10 @@ bool module_t::tdata1_write(processor_t * const proc, unsigned index, const reg_
   reg_t tdata1 = val;
   reg_t tdata2 = triggers[index]->tdata2_read(proc);
 
+  // dmode only writable from debug mode
+  if (!proc->get_state()->debug_mode)
+    tdata1 = set_field(tdata1, CSR_TDATA1_DMODE(xlen), 0);
```

Comment:
There is an unwritten assumption here that `CSR_TDATA1_DMODE(xlen)` and `MCONTROL_DMODE(xlen)` and `CSR_ITRIGGER_DMODE(xlen)` and `CSR_ETRIGGER_DMODE(xlen)` are all equal. Should we have assertion(s) to that effect?